### ISSUE TITLE
Possible doc bug in docker-and-lets-encrypt instructions

### ISSUE THREAD

**Issue by hartzell** [containous/traefik#3105]
### Do you want to request a *feature* or report a *bug*?

bug (though perhaps it's a beginner mistake)

### What did you do?

Tried to set up a server using Lets Encrypt following the [docker-and-lets-encrypt.md](https://github.com/containous/traefik/blob/master/docs/user-guide/docker-and-lets-encrypt.md) instructions.

It didn't work until I appended an 

```toml

[[acme.domains]]
  main = "demo.palomarin.com"
```

to the `traefik.toml` file.

The hostname for the system is set to:

```
[hartzell@demo traefik]$ hostname
demo.palomarin.com
[hartzell@demo traefik]$
```

I'm wondering if the docs are out of date or if I have something else configured incorrectly (or if I'm missing something...).

### What did you expect to see?

I expected that traefik would request a cert for the `hostname` and HTTPS requests to that domain would use that cert.

### What did you see instead?

The default traefik cert was used instead.

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
[hartzell@demo traefik]$ docker run traefik:1.5.4 version
Version:      v1.5.4
Codename:     cancoillotte
Go version:   go1.9.4
Built:        2018-03-15_01:35:21PM
OS/Arch:      linux/amd64
[hartzell@demo traefik]$
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

Working on a Digital Ocean droplet running CentOS Linux release 7.4.1708 (Core).

#### docker-compose.yml

```yaml
version: '2'

services:
  traefik:
    image: traefik:1.5.4
    restart: always
    ports:
      - 80:80
      - 443:443
    networks:
      - web
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - /opt/traefik/traefik.toml:/traefik.toml
      - /opt/traefik/acme.json:/acme.json
    container_name: traefik

  whoami:
    image: emilevauge/whoami #A container that exposes an API to show it's IP address
    labels:
      - "traefik.frontend.rule=Path:/fluffy"
      - "traefik.enable=true"
    networks:
      - web

networks:
  web:
    external: true
```

#### traefik.toml

```toml
debug = true

logLevel = "ERROR"
defaultEntryPoints = ["https","http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
    [entryPoints.http.redirect]
    entryPoint = "https"
  [entryPoints.https]
  address = ":443"
    [entryPoints.https.tls]

[retry]

[docker]
endpoint = "unix:///var/run/docker.sock"
domain = "demo.palomarin.com"
watch = true
exposedbydefault = false

[acme]
email = "hartzell@alerce.com"
storage = "acme.json"
entryPoint = "https"
OnHostRule = true
[acme.httpChallenge]
entryPoint = "http"

# These seem to be required to make the certs work...
# [[acme.domains]]
#   main = "demo.palomarin.com"
```

**Comment by jakeprem** [comment]
I have the same issue. I'm currently testing to see if adding `[[acme.domains]]` fixes it.

**Comment by nmengin** [comment]
Hello @hartzell ,

Many thanks for your interest in the project.

I guess there is a misunderstood about the word `hostname`.
In the context of Træfik, it is not your machine hostname but the host name provided by the incoming HTTP request.

For example, in the request `https://demo.palomarin.com/fluffy`, the hostname is `demo.palomarin.com`.

Træfik will generate automatically an ACME certificate if you specify a backend with an `Host` rule.

In your example, if you replace the label `- "traefik.frontend.rule=Path:/fluffy"` by `- "traefik.frontend.rule=Host:fluffy.demo.palomarin.com"`, a certificate will be generated for the domain `fluffy.demo.palomarin.com`.

I hope I answered your question.

I close the issue but feel free to reopen it if necessary or to join our [Slack workspace](https://traefik.herokuapp.com/) for more [community #support](https://traefik.slack.com/messages/C0CDT22PJ/).

**Comment by hartzell** [comment]
@nmengin -- **Thanks!**

If I understand your explanation correctly, Traefik will generate ACME certificates for any hostname's mentioned in Host rules.  Because I was using only Path rules (no Host rules) I needed to explicitly use the `acme.domains` statement.  I was *incorrectly* assuming that Traefik would generate a default certificate using the machine's hostname.

Is that a correct summary?

**Comment by hartzell** [comment]
I've verified this through experimentation.  Thanks!

**Comment by jakeprem** [comment]
I ended up fixing my issue by switching to a newer version of the Traefik image.

The example at https://docs.traefik.io/user-guide/docker-and-lets-encrypt/  shows using traefik:1.3.5, updating to 1.5.4 has things working.

My issue was caused by/is related to #1832 and #2691, with error messages `Could not determine solvers` and `Error getting ACME certificates`. Another symptom was seeing the Traefik default certificates listed in the browser when it tagged applications that Traefik was proxying to as insecure.

(Hopefully that's enough detail for Google to link back here if someone else has the same problem)
